Documentation/contributing/testing/unit.rst

```diff
@@ -123,3 +119,121 @@ run this in a terminal next to your editor:
     $ watchtest pkg/policy
 
 This shell script depends on the ``inotify-tools`` package on Linux.
+
+.. _migrate-gocheck:
+
+Migrating tests off of ``gopkg.in/check.v1``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The Cilium codebase has historically used ``gopkg.in/check.v1`` to write unit and
+integration tests. Most of the functionality offered by that package is now
+present in `testing`_. What ``gocheck`` calls checkers are now available in `assert`_:
```

Comment:
More a question than a comment. Is it there any guideline about [*assert*](https://pkg.go.dev/github.com/stretchr/testify/assert) vs [*require*](https://pkg.go.dev/github.com/stretchr/testify/require)? It seems that both are used in the repo currently, but this doc mentions *assert* only.

Reply:
Thanks for the clarification, makes definitely sense to me. I personally tend to mostly use *require* to be on the safe side, given that in many cases an error would cause a cascade of failures. But since here I saw that only *assert* was mentioned, I started wondering whether there was any different guideline...

Reply:
Yeah, the interfaces are the same, use whatever makes the most sense for the situation you're in.

Reply:
I'm probably in the require camp myself, didn't realise that assert didn't `Fatal` the test. I'll add a little blurb.

Reply:
Turns out that `C.Assert` behaves like `require.*`, so I updated the doc accordingly.